### PR TITLE
Dit 2457

### PR DESCRIPTION
**Pull request by ansahdavid**

### CODE REVIEW

File: app/controllers/v2/MyCasesController.scala [hunk 1]
```diff
@@ -41,35 +42,54 @@ class MyCasesController @Inject() (
...
+          events =>
+            val eventsLatestFirst = events.results.sortBy(_.timestamp)(Event.latestFirst)
+            val latestCompletedEvent = eventsLatestFirst.collectFirst {
+              case event @ Event(_, _, _, caseReference, _) => Map(caseReference -> event)
+              case _                                                                 => Map.empty
+            }
+            latestCompletedEvent.getOrElse(Map.empty)
+        }
+      }
+      .map(_.foldLeft(Map.empty[String, Event])(_ ++ _))
```
Comment (DavidGregory084) [inline app/controllers/v2/MyCasesController.scala:79]: Oh yeah I forgot - did you try using the `findCompletionEvents` function we added here? No biggie but might improve performance

File: app/controllers/v2/MyCasesController.scala [hunk 2]
```diff
@@ -41,35 +42,54 @@ class MyCasesController @Inject() (
...
 
   def displayMyCases(activeSubNav: SubNavigationTab = AssignedToMeTab): Action[AnyContent] =
     (verify.authenticated andThen verify.mustHave(Permission.VIEW_MY_CASES)).async {
       implicit request: AuthenticatedRequest[AnyContent] =>
         for {
           cases <- casesService.getCasesByAssignee(request.operator, NoPagination())
           referralEventsByCase <- getReferralEvents(cases)
+          completeEventsByCase <- getCompletedEvents(cases)
+          _ = logger.info(completeEventsByCase.toString())
```
Comment (DavidGregory084) [inline app/controllers/v2/MyCasesController.scala:88]: Can we remove this logging before merging?

**Review by DavidGregory084 — APPROVED** [review]
Looks great 👍

File: app/controllers/v2/MyCasesController.scala (hunk 1 above)
Comment (ansahdavid) [reply app/controllers/v2/MyCasesController.scala:79]: Good point David and have now sorted this, thanks.

File: app/controllers/v2/MyCasesController.scala (hunk 2 above)
Comment (ansahdavid) [reply app/controllers/v2/MyCasesController.scala:88]: Yep removed thanks

**Pull request closed without merging**